pkg/pcr/replay_test.go

```diff
@@ -0,0 +1,122 @@
+package pcr
+
+import (
+	"testing"
+
+	"github.com/9elements/converged-security-suite/v2/pkg/tpmeventlog"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReplay(t *testing.T) {
+	makeMeasurementDigest := func(alg tpm2.Algorithm, lastByteValue uint8) []byte {
+		h, err := alg.Hash()
+		require.NoError(t, err)
+		r := make([]byte, h.Size())
+		r[len(r)-1] = lastByteValue
+		return r
+	}
+
+	makeFinalDigest := func(alg tpm2.Algorithm, lastByteValues []uint8) []byte {
+		h, err := alg.Hash()
+		require.NoError(t, err)
+		final := make([]byte, h.Size())
+		final[len(final)-1] = lastByteValues[0]
+		extend := func(d []byte) {
+			hasher := h.HashFunc().New()
+			_, err = hasher.Write(final)
+			require.NoError(t, err)
+			_, err = hasher.Write(d)
+			require.NoError(t, err)
+			final = hasher.Sum(nil)
+		}
+		single := make([]byte, h.Size())
+		for _, lastByteValue := range lastByteValues[1:] {
+			single[len(single)-1] = lastByteValue
+			extend(single)
+		}
+		return final
+	}
+
+	eventLog := &tpmeventlog.TPMEventLog{
```

Comment:
It would be good to just have an ASCII table of the events we're expecting in this event log, as a bit easier to read and digest in case we need to go back and change things

Reply:
Fixed.